url354ce1a2d1_dagaren_gladiator_chess_tgz-pJ8-es_dagaren_gladiator_main_MainJ8/src/es/dagaren/gladiator/representation/BitboardPosition.java

```diff
@@ -0,0 +1,2907 @@
+      if(keyValue != null)
+      {
+         keyValue = keyValue - 1;
+         if(keyValue == 0)
+            gamePositionsHash.remove(zobristKey.getKey());
+         else
+            gamePositionsHash.put(zobristKey.getKey(), keyValue);
+      }
+      
+      Piece sp = mov.getSourcePiece();
+      Piece dp = mov.getDestinationPiece();
+      Square ss = mov.getSource();
+      Square ds = mov.getDestination();
+      
+      removePiece(ds);
+      if(dp != null)
+         putPiece(dp,ds);
+      putPiece(sp,ss);
+      
+      //Si es captura al paso se pone un peón en la casilla correspondiente 
+      //TODO añadir comprobación de si es casilla al paso en el caso
+      //de que sea un movimiento de peón en diagonal (origen.file != destino.file)
+      //y en la casilla destino no exista pieza
+      if(mov.isInPassant())
+      {
+         if(ds.getRank() == Rank._3)
+         {
+            putPiece(Piece.WHITE_PAWN, enPassantSquareSource[ds.index]);
+         }
+         else if(ds.getRank() == Rank._6)
+         {
+            putPiece(Piece.BLACK_PAWN, enPassantSquareSource[ds.index]);
+         }
+      }
+      
+      if(sp == Piece.WHITE_KING && ss == Square.e1)
+      {
+         if(ds == Square.g1)
+         {
+            //Enroque corto blanco
+            removePiece(Square.f1);
+            putPiece(Piece.WHITE_ROOK,Square.h1);
+            
+         }
+         else if(ds == Square.c1)
+         {
+            //Enroque largo blanco
+            removePiece(Square.d1);
+            putPiece(Piece.WHITE_ROOK,Square.a1);
+         }
+      }
+      else if(sp == Piece.BLACK_KING && ss == Square.e8)
+      {
+         if(ds == Square.g8)
+         {
+            //Enroque corto negro
+            removePiece(Square.f8);
+            putPiece(Piece.BLACK_ROOK,Square.h8);
+         }
+         else if(ds == Square.c8)
+         {
+            //Enroque largo negro
+            removePiece(Square.d8);
+            putPiece(Piece.BLACK_ROOK,Square.a8);
+         }
+      }
+      
+      //Actualizar turno
+      this.setTurn(turn.opposite());
+      
+      //Se actualizan los bitboards para la posición resultante
+      updateAttackedFromFull();
+      
+      //Se recoge el estado irreversible de la pila y se reinstaura
+      IrreversibleState state = stateStack.removeLast();
+      this.setEnPassantSquare(state.enPassantSquare);
+      this.setFiftyMovesRuleIndex(state.fiftyMovesCounter);
+      this.setCastlingLong(Colour.WHITE, state.longCastling[Colour.WHITE.index]);
+      this.setCastlingLong(Colour.BLACK, state.longCastling[Colour.BLACK.index]);
+      this.setCastlingShort(Colour.WHITE, state.shortCastling[Colour.WHITE.index]);
+      this.setCastlingShort(Colour.BLACK, state.shortCastling[Colour.BLACK.index]);
+      /////
+      
+      return true;
+   }
+   
+   /* (non-Javadoc)
+    * @see es.dagaren.chesi.Posicion#hacerMovement(es.dagaren.chesi.Movement)
+    */
+   public boolean doMove(Movement mov)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 41. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 1504-1511 into a separate method.